src/agent/agent.c

```diff
@@ -391,44 +394,70 @@ bool agent_set_name(Agent *agent, const char *name) {
 }
 
 bool agent_parse_config(Agent *agent, const char *configfile) {
-        _cleanup_config_ _config *config = NULL;
-        topic *topic = NULL;
-        const char *name = NULL, *host = NULL, *port = NULL;
+        int result = 0;
+
+        result = cfg_initialize(&agent->config);
+        if (result != 0) {
+                fprintf(stderr, "Error initializing configuration: '%s'.", strerror(result));
+                return false;
+        }
+
+        result = cfg_load_complete_configuration(
+                        agent->config,
+                        NULL, // TODO: https://github.com/containers/hirte/issues/147
+                        CFG_ETC_HIRTE_AGENT_CONF,
+                        NULL); // TODO: https://github.com/containers/hirte/issues/148
+        if (result != 0) {
+                fprintf(stderr, "Error loading configuration: '%s'.", strerror(result));
+                cfg_dispose(agent->config);
+                return false;
+        }
 
-        config = parsing_ini_file(configfile);
-        if (config == NULL) {
+        if (configfile != NULL) {
+                result = cfg_load_from_file(agent->config, configfile);
+                if (result != 0) {
+                        fprintf(stderr,
+                                "Error loading configuration file '%s', error code '%s'.",
+                                configfile,
+                                strerror(result));
+                        cfg_dispose(agent->config);
+                        return false;
+                }
+        }
+
+        result = cfg_set_default_section(agent->config, CFG_SECT_AGENT);
```

Comment:
Shouldn't this be set before loading the files?

Reply:
I guess that doesn't really matter.

Reply:
This is used only in `cfg_get_value` and `cfg_set_value`, which are not called during parsing files or environment, but it needs to be called before accessing existing configuration